front/app/containers/App/index.tsx

```diff
@@ -104,22 +108,20 @@ const App = ({ children }: Props) => {
         appConfiguration.data.attributes.settings.core.timezone
       );
 
-      // Dynamically load Moment.js locales
-      const localeImports = import.meta.glob(
-        '/node_modules/moment/locale/*.js'
-      );
-
       uniq(
         appConfiguration.data.attributes.settings.core.locales
-          .filter((locale) => locale !== 'en')
-          .map((locale) => appLocalesMomentPairs[locale])
-      ).forEach((locale) => {
-        const localePath = `/node_modules/moment/locale/${locale}.js`;
-        const loadLocale = localeImports[localePath];
-
-        // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
-        if (loadLocale) {
-          loadLocale();
+          .filter((loc) => loc !== 'en')
+          .map((loc) => appLocalesMomentPairs[loc])
+      ).map(async (loc) => {
+        try {
+          // Dynamically import the locale only if it matches the current locale.
+          // This ensures we only load the required locale when needed.
+          // If the locale changes, the appropriate one will be imported in some other code.
+          if (loc === locale) {
```

Comment:
I just realised this will never be true. loc is a moment locale, which means it's lowercase, and our locales have an uppercase part.

Reply:
Using more descriptive variable names here can help spot this (`momentLocale` instead of `loc`).